pkg/controller/datavolume-controller.go

```diff
@@ -614,6 +616,30 @@ func newPersistentVolumeClaim(dataVolume *cdiv1.DataVolume) (*corev1.PersistentV
 
 	annotations := make(map[string]string)
 
+	if dataVolume.Spec.ResizeTo != "" || dataVolume.Spec.ResizeTo == "max" {
+		var resizeToRequested resource.Quantity
+		var resizeTo int64
+		pvcSize := dataVolume.Spec.PVC.Resources.Requests[corev1.ResourceStorage]
+		// calculate 5% overhead for filesystem
+		overhead := resource.NewQuantity(int64(float64(pvcSize.Value())*0.05), pvcSize.Format)
+		pvcSize.Sub(*overhead)
+		if dataVolume.Spec.ResizeTo == "max" {
+			resizeToRequested = pvcSize
+			resizeTo = pvcSize.Value()
+			glog.V(1).Infof("Downloaded image will be resized to max available value of %s.", pvcSize.String())
+		} else {
+			resizeToRequested = resource.MustParse(dataVolume.Spec.ResizeTo)
+			if pvcSize.Value() >= resizeToRequested.Value() {
+				resizeTo = resizeToRequested.Value()
+				glog.V(1).Infof("Downloaded image will be resized to %s.", resizeToRequested.String())
+			} else {
+				resizeTo = pvcSize.Value()
+				glog.V(1).Infof("Can't resize image to %s. Image can by only resized up to %s", resizeToRequested.String(), pvcSize.String())
+			}
+		}
+		annotations[AnnResizeTo] = fmt.Sprint(resizeTo, "B")
+	}
+
```

Comment:
I think the following is shorter and more readable (I did not check it, but you get the idea). Let me know if you disagree.

if dataVolume.Spec.ResizeTo != "" {
		var newSize int64
		pvcSize := dataVolume.Spec.PVC.Resources.Requests[corev1.ResourceStorage]
		// calculate 5% overhead for filesystem
		overhead := resource.NewQuantity(int64(float64(pvcSize.Value())*0.05), pvcSize.Format)
		pvcSize.Sub(*overhead)
		resizeTo = resource.MustParse(dataVolume.Spec.ResizeTo)
		
		if dataVolume.Spec.ResizeTo == "max" || resizeTo.Value() > pvcSize.Value()   {
			glog.V(1).Infof("Downloaded image will be resized to max available value of %s.", pvcSize.String())			
			newSize = pvcSize.Value()
		} else { 
			glog.V(1).Infof("Downloaded image will be resized to %s.", resizeTo.String())			
			newSize = resizeTo.Value()
		}	
		annotations[AnnResizeTo] = fmt.Sprint(newSize, "B")
}